support-frontend/app/actions/AsyncAuthenticatedBuilder.scala

```diff
@@ -33,7 +34,10 @@ class AsyncAuthenticatedBuilder[U](
     userinfo(request)
       .flatMap(user => block(new AuthenticatedRequest(user, request)))
       .recover { case err =>
-        logger.error("unable to authorize user", err)
+        // User shouldn't necessarily be signed in,
+        // therefore, don't log failure to authenticate as a result of this with log level ERROR.
+        if (isUserNotSignedInError(err)) logger.info(s"unable to authorize user - $err")
+        else logger.error(s"unable to authorize user - $err")
```

Comment:
did you consider pattern matching?
```scala
err match {
  case err @ UserCredentialsMissingError => logger.info(s"unable to authorize user - $err")
  case other => logger.error(s"unable to authorize user - $err")
}
```

Reply:
as in the following? have you got a preference?
```scala
.recover { case err =>
  // User shouldn't necessarily be signed in,
  // therefore, don't log failure to authenticate as a result of this with log level ERROR.
  err match {
    case _: UserCredentialsMissingError => SafeLogger.info(s"unable to authorize user - $err")
    case _ => SafeLogger.error(scrub"unable to authorize user", err)
  }
  
  onUnauthorized(request)
}
```

Reply:
yes that's what I meant, purely to avoid splitting out something that doesn't have a meaning independent of the actual `UserCredentialsMissingError` itself, and to make it more type safe (I am pretty sure scala will complain if the type of "err" (Throwable?) is not a valid superclass of UserCredentialsMissingError etc)

Reply:
Ok, so have used pattern matching and de-deduped the error logging logic by extracting it out to a utility method (which uses the `SafeLogger`).